distributed/shuffle/shuffle_extension.py

```diff
@@ -68,43 +78,131 @@ def npartitions_for(self, worker: str) -> int:
 class Shuffle:
     "State for a single active shuffle"
 
-    def __init__(self, metadata: ShuffleMetadata, worker: Worker) -> None:
+    def __init__(
+        self,
+        metadata: ShuffleMetadata,
+        worker: Worker,
+        executor: ThreadPoolExecutor,
+    ) -> None:
         self.metadata = metadata
         self.worker = worker
-        self.output_partitions: defaultdict[int, list[pd.DataFrame]] = defaultdict(list)
+        self.executor = executor
+
+        import pyarrow as pa
+
+        self.multi_file = MultiFile(
+            dump=functools.partial(
+                dump_batch, schema=pa.Schema.from_pandas(self.metadata.empty)
+            ),
+            load=load_arrow,
+            directory=os.path.join(self.worker.local_directory, str(self.metadata.id)),
```

Comment:
from an API perspective, I'm wondering if the MultiFile shouldn't be responsible for creating the directory. It is also responsible for cleaning it up